### PR TITLE
add red-hat-openshift script

### PR DESCRIPTION
**Pull request by dadav**
This adds a script to get release informations for https://endoflife.date/red-hat-openshift

The docs repo is the source for this information:
https://github.com/openshift/openshift-docs

Each major release has a dedicated branch and a release file (for example release_notes/ocp-4-10-release-notes.adoc). A regex expression is used to extract the versions and dates.

The repository is pretty large, so I tried to keep the fetch operations to a minimum.

### CODE REVIEW

**Comment by marcwrobel** [comment]
@captn3m0, do you think the change you made in https://github.com/endoflife-date/release-data/pull/120#issuecomment-1632273872 could have break the CI on PR ?

**Comment by captn3m0** [comment]
No, it only runs on push events to main repo, not on fork PRs.

**Comment by dadav** [comment]
Maybe the problem is that https://github.com/endoflife-date/release-data/blob/main/.github/workflows/update.yml#L2 is missing *pull_request*? I can't find the related action, it seems to waiting for it...

**Comment by marcwrobel** [comment]
I think you are right. Anyway we are able to validate the script locally, so shouldn't be an issue.

File: src/red-hat-openshift.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+    def clone(self):
+        self.repo_dir.mkdir(parents=True, exist_ok=True)
+        if not exists(f"{self.repo_dir}/.git"):
+            self.run("init")
+            self.run(f"remote add origin {self.url}")
+
+    def list_branches(self):
+        raw = self.run(
+            "ls-remote origin 'refs/heads/enterprise-[4-9]*'",
```
Comment (marcwrobel) [inline src/red-hat-openshift.py:56]: Why limiting to OpenShift 4+ ?

**Review by marcwrobel — APPROVED** [review]
Great work @dadav !

I think I will extract the Git class so it can be mutualized the `debian.py`.

**Comment by marcwrobel** [comment]
@captn3m0, could merge this one (the required check must be disabled temporarily to do it, see https://github.com/endoflife-date/release-data/pull/119#issuecomment-1633686404).

File: src/red-hat-openshift.py (hunk 1 above)
Comment (dadav) [reply src/red-hat-openshift.py:56]: They changed the filename schema from `ocp_{version}_release_notes.adoc` to `ocp-{version}-release-notes.adoc`. OpenShift 1-3 won't get any updates anymore, so it doesn't seemed to make sense to support that..

Comment (dadav) [reply src/red-hat-openshift.py:56]: Also, the content seems to be different in the old versions.

Comment (marcwrobel) [reply src/red-hat-openshift.py:56]: Ok, thanks for the precision. Could be interesting to leave a comment about that in the script.

Comment (dadav) [reply src/red-hat-openshift.py:57]: good idea, added 05d22b601222bcca422fff07bcec192c06d30e83

**Comment by dadav** [comment]
> I think I will extract the Git class so it can be mutualized the `debian.py`.

Ok, will work on that :)

File: src/red-hat-openshift.py (hunk 1 above)
Comment (marcwrobel) [reply src/red-hat-openshift.py:57]: Thanks !

**Comment by dadav** [comment]
Should be fine now. I ran both scripts and they work as expected.

**Comment by marcwrobel** [comment]
> @captn3m0, could you merge this one (the required check must be disabled temporarily to do it, see [#119 (comment)](https://github.com/endoflife-date/release-data/pull/119#issuecomment-1633686404)).

The Update Data workflow is not required anymore (see #121). I can do the merge myself now.

**Review by marcwrobel — APPROVED** [review]
That's a really nice PR @dadav, thanks !

Just checked the PR and executed both the debian and openshift scripts. LGTM, merging.

**Pull request merged**